src/common/utils.ts

```diff
@@ -191,7 +191,12 @@ export const getDisplayNameForSearch = (conversation: any, groups?: boolean) =>
   if (groups) {
     displayName = conversation.name || conversation.waGroup?.label || '';
   } else {
-    displayName = conversation.name || conversation.maskedPhone || conversation.contact?.name || '';
+    displayName =
+      conversation?.name ||
+      conversation?.maskedPhone ||
+      conversation?.contact?.maskedPhone ||
+      conversation?.contact?.name ||
```

Comment:
Let's pass the contact object only inside this function rather than checking for maskedPhone and name again inside contact.